src/Build/BuildCheck/Infrastructure/EditorConfig/EditorConfigGlobsMatcher.cs

```diff
@@ -0,0 +1,615 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// Note:
+// Copied from https://github.com/dotnet/roslyn/blob/06d3f153ed6af6f2b78028a1e1e6ecc55c8ff101/src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.SectionNameMatching.cs
+// with some changes to make it quicker to integrate into the MSBuild.
+// Changes:
+//  1. ArrayBuilder was replaced with List.
+//  2. Exceptions. Wrap in try/catch blocks for proper reporting
+
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.BuildCheck.Infrastructure.EditorConfig
+{
+    internal class EditorConfigGlobsMatcher
+    {
+        internal readonly struct SectionNameMatcher
+        {
+            private readonly ImmutableArray<(int minValue, int maxValue)> _numberRangePairs;
+            // public for testing
+            public Regex Regex { get; }
```

Comment:
Could we make this internal for testing instead?

Reply:
Agree! Thank you!  updated